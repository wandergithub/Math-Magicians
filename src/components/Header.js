import { Link } from 'react-router-dom';
import './styles/Header.css';

const Header = () => (
  <header className="nav-bar">
    <h1>Math Magicians</h1>
    <div className="link-container">
      <Link to="/">Home</Link>
      <Link to="/Calculator" className="mid-b">Calculator</Link>
      <Link to="/Quote">Quote</Link>
    </div>
  </header>
);

export default Header;
