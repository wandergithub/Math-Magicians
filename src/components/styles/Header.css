.nav-bar {
  display: flex;
  justify-content: space-between;
  align-content: center;
  margin: 20px;
  box-shadow: 5px 10px 8px 10px #8b7f78a0;
  border-radius: 10px;
}

.nav-bar h1 {
  margin-left: 10px;
}

.link-container {
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 10px;
  margin-right: 10px;
}

.link-container > * {
  text-decoration: none;
  color: black;
}

.mid-b {
  border-right: 1px solid black;
  border-left: 1px solid black;
  padding: 0 5px;
}
