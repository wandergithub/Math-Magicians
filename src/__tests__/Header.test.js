import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Header from '../components/Header';

describe("Header component tests", () => {
  it("Renders correctly", () => {
    const tree = renderer 
      .create(
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
