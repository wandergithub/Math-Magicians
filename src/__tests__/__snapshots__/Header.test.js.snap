// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component tests Renders correctly 1`] = `
<header
  className="nav-bar"
>
  <h1>
    Math Magicians
  </h1>
  <div
    className="link-container"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="mid-b"
      href="/Calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
    <a
      href="/Quote"
      onClick={[Function]}
    >
      Quote
    </a>
  </div>
</header>
`;
